src/main/java/org/folio/tlib/postgres/impl/PgCqlQueryImpl.java

```diff
@@ -0,0 +1,270 @@
+      case ">":
+      case "<":
+      case "<=":
+      case ">=":
+        return base;
+      default:
+        throw new IllegalArgumentException("Unsupported operator " + base + " for: "
+            + termNode.toCQL());
+    }
+  }
+
+  /**
+   * See if this is a CQL query with a existence check (NULL or NOT NULL).
+   * <p>Empty term makes "IS NULL" for CQL relation =, "IS NOT NULL" for CQL relation <>.
+   * </p>
+   * @param field CQL field.
+   * @param termNode term.
+   * @return SQL op for NULL; null if not a NULL check.
+   */
+  String handleNull(PgCqlField field, CQLTermNode termNode) {
+    if (!termNode.getTerm().isEmpty()) {
+      return null;
+    }
+    String base = termNode.getRelation().getBase();
+    switch (base) {
+      case "==":
+        return null;
+      case "=":
+        return field.getColumn() + " IS NULL";
+      case "<>":
+        return field.getColumn() + " IS NOT NULL";
+      default:
+        throw new IllegalArgumentException("Unsupported operator " + base + " for: "
+            + termNode.toCQL());
+    }
+  }
+
+  String handleTypeUuid(PgCqlField field, CQLTermNode termNode) {
+    String s = handleNull(field, termNode);
+    if (s != null) {
+      return s;
+    }
+    UUID id = UUID.fromString(termNode.getTerm()); // so IllegalArgumentException is thrown
+    String pgTerm = "'" + id + "'";
+    String op = basicOp(termNode);
+    return field.getColumn() + op + pgTerm;
+  }
+
+  String handleTypeText(PgCqlField field, CQLTermNode termNode) {
+    String s = handleNull(field, termNode);
+    if (s != null) {
+      return s;
+    }
+    String base = termNode.getRelation().getBase();
+    boolean ft = "=".equals(base) || "all".equals(base);
+    String cqlTerm = termNode.getTerm();
+    StringBuilder pgTerm = new StringBuilder();
+    boolean backslash = false;
+    for (int i = 0; i < cqlTerm.length(); i++) {
+      char c = cqlTerm.charAt(i);
+      if (c == '\'' && !backslash) {
+        pgTerm.append('\''); // important to avoid SQL injection
+      } else if (c == '*' && ft) {
+        if (!backslash) {
+          throw new IllegalArgumentException("Masking op * unsupported for: " + termNode.toCQL());
+        }
+      } else if (c == '?' && ft) {
+        if (!backslash) {
+          throw new IllegalArgumentException("Masking op ? unsupported for: " + termNode.toCQL());
+        }
+      } else if (c == '^' && ft) {
+        if (!backslash) {
+          throw new IllegalArgumentException("Anchor op ^ unsupported for: " + termNode.toCQL());
+        }
+      } else if (backslash) {
+        pgTerm.append('\\'); // pass-tru the backslash for Postgres to honor (including \')
+      }
+      if (c == '\\') {
+        backslash = true;
+      } else {
+        backslash = false;
+        pgTerm.append(c);
+      }
+    }
+    if (backslash) {
+      pgTerm.append('\\');
+    }
+    if (!ft) {
+      return field.getColumn() + " " + basicOp(termNode) + " E'" + pgTerm + "'";
```

Comment:
I'll just have to trust this line and the next one, not knowing enough PostgreSQL to comment.

Reply:
ok.